### PR TITLE
Update spaCy to 3.0

### LINKED ISSUE #4950: Support spaCy v3
spaCy released the version v3, with great improvements. https://github.com/explosion/spaCy/releases/tag/v3.0.0
Unfortunately allennlp is limiting to spacy<2.4 and the dependency resolver do not allow allennlp and spaCy v3 to be used in the same project.

Would be great to bump the requirement.

### PR DESCRIPTION
**Pull request by bratao**
Fixes #4950 .

Changes proposed in this pull request:

- Update to spaCy 3.0

### CODE REVIEW

File: setup.py
Comment (epwalsh) [inline setup.py:58]: Did you mean

```suggestion
        "spacy>=2.1.0,<3.1",
```

**Comment by bratao** [comment]
Dependabot did a better job

**Comment by epwalsh** [comment]
@bratao I did see there were some CI failures in the dependabot PR. If you want to attempt to fix those issues feel free to re-open this PR since you won't be able to commit to dependabot's branch. Otherwise myself or another maintainer will try to get to that when we can.

**Comment by bratao** [comment]
Sure, let´s do it

**Comment by epwalsh** [comment]
Thanks @bratao! Have you looked at the test that's failing on models?

image

You are probably more familiar with SpaCy than I am. Do you know how to fix this? It looks like `TAG_MAP` was totally removed from `lang/en` (https://github.com/explosion/spaCy/tree/master/spacy/lang/en).

**Comment by bratao** [comment]
@epwalsh Yeah, from what I understand it was completely removed from v3. 
I see that only constituency_parser.py uses it, for build a representation. How would you like to solve it? Maybe we can import the old tag_set and past on constituency_parser.py

**Comment by epwalsh** [comment]
> Maybe we can import the old tag_set and copy&paste on constituency_parser.py

Hey @bratao, that sounds like a good solution

File: .github/workflows/ci.yml
Comment (epwalsh) [inline .github/workflows/ci.yml:155]: TODO: remove this before merging

```suggestion
```

**Review by epwalsh — APPROVED** [review]

**Review by epwalsh — APPROVED** [review]
Thanks @bratao!

**Pull request merged**
